File: puzzles/js/onboarding/onboarding.js
/**
 * CodinGame planet is being attacked by slimy insectoid aliens.
 * <---
 * Hint:To protect the planet, you can implement the pseudo-code provided in the statement, below the player.
 **/

// game loop
while (true) {
  let enemy1 = readline(); // name of enemy 1
  let dist1 = parseInt(readline()); // distance to enemy 1
  let enemy2 = readline(); // name of enemy 2
  let dist2 = parseInt(readline()); // distance to enemy 2
  
  if (dist1 < dist2) {
    console.log(enemy1);
  } else {
    console.log(enemy2);
  }
}

File: puzzles/kotlin/src/onboarding/onboarding.kt
import java.util.Scanner

fun main(args: Array<String>) {
    val scanner = Scanner(System.`in`)
    while (true) {
        val enemy1 = scanner.next() // name of enemy 1
        val dist1 = scanner.nextInt() // distance to enemy 1
        val enemy2 = scanner.next() // name of enemy 2
        val dist2 = scanner.nextInt() // distance to enemy 2

        if (dist1 < dist2) {
            println(enemy1)
        } else {
            println(enemy2)
        }
    }
}
